docs/platform/configuration.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Configuration API
+description: The Configuration API provides a means for types to be configured by third-parties.
+parent-title: Pentaho Platform JavaScript APIs
+parent-path: ../..
+layout: default
+---
+
+The 
+[Configuration API]({{site.refDocsUrlPattern | replace: '$', 'pentaho.config'}}) 
+provides a means for _types_ to be configured by third-parties.
+
+**Types** are known by their _string_ identifier and are, for all other purposes, opaque entities
+— these may or may not exist as actual JavaScript classes; these may simply represent an interface type.
+
+**Configurations** are JavaScript objects that conform to the 
+[`pentaho.config.spec.IRuleSet`]({{site.refDocsUrlPattern | replace: '$', 'pentaho.config.spec.IRuleSet'}}) interface
+— essentially, a set of configuration rules,
+[`pentaho.config.spec.IRule`]({{site.refDocsUrlPattern | replace: '$', 'pentaho.config.spec.IRule'}}).
+Typically, 
+configurations are provided as the value returned by an AMD/RequireJS module.
+This module needs to be advertised to the configuration system by registering it
+with [`pentaho/service`]({{site.refDocsUrlPattern | replace: '$', 'pentaho.service'}}),
+as a service of type `pentaho.config.spec.IRuleSet`.
+
+**Configuration Rules** are composed of the following parts:
+
+1. The [**select**]({{site.refDocsUrlPattern | replace: '$', 'pentaho.config.spec.IRule#select'}}) object
+   specifies the targeted _type_ (or _types_) and the values of any 
+   [Pentaho environment variables]({{site.refDocsUrlPattern | replace: '$', 'pentaho.context.IContext'}})
+   to which it applies. Alternative values for a variable can be specified using a JavaScript array. 
+   The most useful environment variable is 
+   [application]({{site.refDocsUrlPattern | replace: '$', 'pentaho.context.IContext#application'}}),
+   as it allows creating rules that are only applied when a _type_ is being used by a certain _application_,
+   like, for example, 
+   [CDF](http://community.pentaho.com/ctools/cdf/) or 
+   [Analyzer](http://www.pentaho.com/product/business-visualization-analytics).
+   See also [Known Values of Pentaho Environment Variables](known-values-of-pentaho-environment-variables).
+   
+2. The [**apply**]({{site.refDocsUrlPattern | replace: '$', 'pentaho.config.spec.IRule#apply'}}) object
+   specifies the actual configuration properties and their values.
+   You will need to consult the reference documentation of the target _type_ to known the list of available properties.
```

Comment:
The _type_ is just something... Not the same thing than a _type of the Type API_. That adds a bunch of confusion... Why `isBrowsable` is at the top but `lineWidth` is inside `props`? (not saying I don't know why, but noting it might be confusing at first look)

I have no suggestions, however.